WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsViewModel.swift

```diff
@@ -45,3 +76,13 @@ extension ProductSettingsViewModel {
         }
     }
 }
+
+// Editable data
+//
+final class ProductSettings {
+    var status: ProductStatus
```

Comment:
unrelated: somehow I was hoping the property (and other properties in the future) could be immutable and this could be a plain `struct` since they are passed around between screens. though I understand it could be a hassle to create a new `ProductSettings` when calling the completion block from each Product Settings screen when the number of properties grows (like `Product`). so it's fine as it is, just sharing some thoughts :)

Reply:
I completely agree with you and I had the same thought, for this reason, I made `ProductSettings` a class. We will see when the Settings implementation finishes if it makes sense to transform the class into `struct`, but I am afraid that the properties will be many.

Reply:
I think we're almost there. For example, in `ProductSettingsRows.Status.handleTap`, instead of this: 

```swift
self.settings.status = selected ?? self.settings.status
onCompletion(self.settings)
```

We can have something like:

```swift
// create immutable copy. 
let settings = self.settings.copy(status: selected)
// tell (probably another new class) store that there is a new `ProductSettings`
productSettingsStore.update(newValue: settings)
```

It's like the Yosemite architecture where there's only a single place to _update_ the `ProductSettings` struct. It should make debugging and tracking changes easier. 

That `copy()` method is something that I'm working on proposing (hopefully soon!).

Anyway, I think with this current architecture, things are easier to reason about and refactor. So, I agree with Paolo that we can defer the immutability change later.

Reply:
This is interesting @shiki because while I was implementing this class, I thought about the `copy()` method that I used in the past to create a new object and pass it as `copy` and not `reference`. In that case, I was using a class that conforms to `NSCopying`. It can be a possible solution. We will see what is the status of `ProductSettings` at the end of all the implementations we need. 👍

Reply:
it'd be really cool if there's a way to copy a struct with a subset of properties updated to the input parameters! we have [`ProductUpdater`](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/Yosemite/Model/Updaters/ProductUpdater.swift#L3) for similar use case for edit product, but [every update implementation](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/Yosemite/Model/Updaters/ProductUpdater.swift#L25-L86) gets so long (since the properties could be nil) that I'm hoping to see a more concise way. Looking forward to reading your proposal @shiki !